tests/test_modules/test_ADOdin/test_odinwriterpart.py

```diff
@@ -1,58 +1,202 @@
+            copy(f, tmp_dir)
+
+        # Create a generator to match the test data
+        xs = LineGenerator("x", "mm", 0.0, 4.0, 3)
+        ys = LineGenerator("y", "mm", 0.0, 4.0, 2)
+        compound = CompoundGenerator([ys, xs], [], [])
+        compound.prepare()
+
+        # Call configure to create the VDS
+        # This should work with relative paths but doesn't due to VDS bug
+        self.o.configure(self.context, 0, 6, compound, formatName='odin123',
+                         fileDir=tmp_dir, fileTemplate="%s.h5")
+
+        # Open the created VDS file and dataset to check values
+        vds_path = os.path.join(tmp_dir, 'odin123.h5')
+        vds_file = h5py.File(vds_path, "r")
+        detector_dataset = vds_file['/entry/detector/detector']
+
+        # Check values at indices 0,0
+        self.assertEquals(detector_dataset[0][0][756][393], 1)
+        self.assertEquals(detector_dataset[0][0][756][394], 1)
+        self.assertEquals(detector_dataset[0][0][756][395], 1)
+        self.assertEquals(detector_dataset[0][0][756][396], 1)
+
+        # Change first index
+        self.assertEquals(detector_dataset[1][0][756][393], 4)
+        self.assertEquals(detector_dataset[1][0][756][394], 4)
+        self.assertEquals(detector_dataset[1][0][756][395], 4)
+        self.assertEquals(detector_dataset[1][0][756][396], 4)
+
+        # Change second index
+        self.assertEquals(detector_dataset[0][2][756][393], 3)
+        self.assertEquals(detector_dataset[0][2][756][394], 3)
+        self.assertEquals(detector_dataset[0][2][756][395], 3)
+        self.assertEquals(detector_dataset[0][2][756][396], 3)
+
+        # Todo there are no gaps in my test data at present:-
+        #  update the test data with Alans Gap fill and fix this
+        # # Check some values near the bottom of image to ensure
+        # # the gaps are there
+        # assert detector_dataset[0][0][1685][1521] == 3
+        # assert detector_dataset[1][0][1685][1521] == 131
+        #
+        # assert detector_dataset[0][0][1516][329] == 109
+        # assert detector_dataset[0][1][1516][329] == 136
+        #
+        # # Check some values in the gaps
+        # assert detector_dataset[0][0][395][1202] == 0
+        # assert detector_dataset[1][0][395][1202] == 0
+
+        # Check detector attributes
+        detector_group = vds_file['/entry/detector']
+        assert detector_group.attrs['axes'] == 'y_set,x_set,.,.'
+        assert detector_group.attrs['signal'] == 'detector'
+        assert detector_group.attrs['y_set_indices'] == '0'
+        assert detector_group.attrs['x_set_indices'] == '1'
+
+        # Check _set datasets
+        # N.B. units are encoded as ASCII in the original file, so come
+        # back as type byte in Python 3
+        stage1_x_set_dataset = vds_file['/entry/detector/x_set']
+        assert stage1_x_set_dataset[0] == 0
+        assert stage1_x_set_dataset[1] == 2
+        assert stage1_x_set_dataset.attrs['units'] == b'mm'
```

Comment:
Is this really a bytes literal? Does h5py not convert to Python3 string?

Reply:
It seems I lied about Python 3 tests working. In Python3 this is str but in Python2 it is bytes. Will change the test to case the value to str.